lib/internal/fs/promises.js

```diff
@@ -301,9 +307,11 @@ class FileHandle extends EventEmitter {
       );
     }
 
-
     const readFn = FunctionPrototypeBind(this.read, this);
-    const ondone = FunctionPrototypeBind(this[kUnref], this);
+    const ondone = async () => {
+      this[kUnref]();
+      if (options.autoClose) this.close();
```

Comment:
Shouldn't we await here?
```suggestion
      if (options.autoClose) await this.close();
```